enhancements/rhcos/split-rhcos-into-layers.md

```diff
@@ -0,0 +1,428 @@
+full RHCOS rebuild each time). And it also opens the door to more easily ship
+binaries in the OS without necessarily having to package them as RPMs first.
+
+Finally, keeping the CoreOS layer separate from the node layer provides major
+benefits to OKD. First, because it makes it easier to share the derivation process
+for building the node image (today, OKD maintains its own separate definitions).
+Second, because it allows the sharing of CentOS Stream-based bootimages between
+OCP and OKD (today, OKD uses FCOS bootimages). These changes would take OKD one
+step closer to become a true upstream of OCP.
+
+### User Stories
+
+* As an OpenShift engineer, I want the openshift/os repo to work similarly to other OpenShift repos (with a `Containerfile`), so that CI can easily build and test a PR in a cluster and humans too can more easily build and test PRs or e.g. use Cluster Bot.
+
+* As a CoreOS engineer, I want to only have a single stream of RHCOS per RHEL stream, so that I can greatly reduce the amount of streams to maintain (build, test, monitor for failures, flakes, etc...), reduce the number of bootimage bumps, and enforce more consistency across OCP releases that span the same RHEL version.
+
+* As a CoreOS engineer, I want to build RHCOS on top of rhel-bootc, so that I can benefit from support by the wider RHEL organization on those images and collaborate on CI testing and fixing bugs.
+
+* As an OpenShift customer, I want to be able to follow (most of) the image mode for RHEL documentation and use (most of) the same tooling surrounding it when customizing RHCOS, so that I have a consistent experience across image mode for RHEL and RHCOS and a clear understanding of how they relate to each other.
+
+* As an ART engineer, I want to be able to build the OpenShift node image in a similar way to how the other OpenShift components are built, so that I can simplify my tooling and pipelines.
+
+* As a RHEL kernel engineer, I want to be able to debug an OCP kernel bug the same way I do for image mode for RHEL, so that I don't have to learn different ways to do things.
+
+* As an OKD contributor, I want to be able to use CentOS Stream-based boot images, so that I don't have to keep pivoting from Fedora CoreOS boot images into CentOS Stream everytime.
+
+* As an OKD contributor, I want to be able to share node image building processes with OCP, so that I can reduce the maintenance load.
+
+* (Longer-term) As a product manager, I want image mode for RHEL certification and support guarantees to automatically apply to OpenShift clusters.
+
+### Goals
+
+- Build RHCOS/SCOS on top of rhel-bootc/centos-bootc
+- Share efforts, especially CI and docs, with the larger rhel/centos-bootc project
+- Maintain a single RHCOS stream per RHEL stream
+- Build the OpenShift node image as a layered image on top of RHCOS/SCOS
+- Build the OpenShift node image using Prow (in CI) and Konflux (in prod)
```

Comment:
```suggestion
- Introduce no additional reboots in bootstrapping
```
I think reading through things that's the case, we'd just be going from a CoreOS layer without a Kubelet directly to a node layer with kubelet and all MachineConfig applied. Does that mean then that at least a stub MCD still lives in CoreOS Layer?

Reply:
Or if there's no MCD in these images, can we make that more clear?

Reply:
Executed by podman or something that's in the bootc layer or CoreOS layer in this enhancement?

Reply:
Yes, podman is still going to be in the bootimages.

> Introduce no additional reboots in bootstrapping

I think we should be able to pull that off but it will require some design and testing. Worth highlighting [Colin's comment](https://github.com/openshift/enhancements/pull/1637/files#r1631394952) that OKD at least does this already. But for OCP, yes this would be a regression and would add time to the install flow (esp. bare metal).

Reply:
Am I interpreting this discussion correctly that this change will require an additional reboot during install, and so you are not going to accept Scott's proposed goal?

I'm just confused by the statement "we should be able to pull that off" followed by "this would be a regression and would add time to the install flow". But I might just be missing some context.

Reply:
Just noticing that my original suggestion replaced line 105 which was not my intention. I've updated it to just add line 106.

Reply:
I've made it an explicit goal not.